src/category_theory/limits/shapes/strict_initial.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2021 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+
+import category_theory.limits.shapes.terminal
+import category_theory.limits.shapes.binary_products
+import category_theory.epi_mono
+
+/-!
+# Strict initial objects
+
+This file sets up the basic theory of strict initial objects: initial objects where every morphism
+to it is an isomorphism. This generalises a property of the empty set in the category of sets:
+namely that the only function to the empty set is from itself.
+
+We say `C` has strict initial objects if every initial object is strict, ie given any morphism
+`f : A ⟶ I` where `I` is initial, then `f` is an isomorphism.
+Strictly speaking, this says that *any* initial object must be strict, rather than that strict
+initial objects exist, which turns out to be a more useful notion to formalise.
+
+If the binary product of `X` with a strict initial object exists, it is also initial.
+
+To show a category `C` with an initial object has strict initial objects, the most convenient way
+is to show any morphism to the (chosen) initial object is an isomorphism and use
+`has_strict_initial_object_of_initial_is_strict`.
+
+The dual notion (strict terminal objects) occurs much less frequently in practice so is ignored.
+
+## TODO
+
+* Construct examples of this: `Type*`, `Top`, `Groupoid`, simplicial types, posets.
+* Construct the bottom element of the subobject lattice given strict initials.
+* Show cartesian closed categories have strict initials
+
+## References
+* https://ncatlab.org/nlab/show/strict+initial+object
+-/
+
+
+universes v u
+
+namespace category_theory
+namespace limits
+open category
+
+variables (C : Type u) [category.{v} C]
+
+/--
+We say `C` has strict initial objects if every initial object is strict, ie given any morphism
+`f : A ⟶ I` where `I` is initial, then `f` is an isomorphism.
+
+Strictly speaking, this says that *any* initial object must be strict, rather than that strict
+initial objects exist.
+-/
+class has_strict_initial_object : Prop :=
```

Comment:
I wonder if, given the "strictly speaking" comment above, it would be better to name this `has_strict_initial_objects` (i.e. pluralize).

Reply:
Done! I'll make sure CI approves then merge.